src/sql/base/parts/editableDropdown/browser/dropdown.ts

```diff
@@ -166,53 +157,76 @@ export class Dropdown extends Disposable {
 					e.stopPropagation();
 					break;
 				case KeyCode.DownArrow:
-					if (!this._treeContainer.parentElement) {
+					if (!this._isDropDownVisible) {
 						this._showList();
 					}
-					this._tree.domFocus();
-					this._tree.focusFirst();
+					setTimeout(() => {
+						this._selectList.domFocus();
+						this._selectList.focusFirst();
+					}, 0);
 					e.stopPropagation();
 					e.preventDefault();
 					break;
 			}
 		}));
 
-		this._tree = new Tree(this._treeContainer, {
-			dataSource: this._dataSource,
-			filter: this._filter,
-			renderer: this._renderer,
-			controller: this._controller
-		}, { paddingOnRow: false, indentPixels: 0, twistiePixels: 0 });
-
-		const treeTracker = this._register(DOM.trackFocus(this._tree.getHTMLElement()));
-
-		treeTracker.onDidBlur(() => {
-			if (!this._input.hasFocus()) {
-				this._onBlur.fire();
+		this._selectList = new List('EditableDropdown', this._selectListContainer, this, [new SelectListRenderer()], {
+			useShadows: false,
+			verticalScrollMode: ScrollbarVisibility.Visible,
+			keyboardSupport: true,
+			mouseSupport: true,
+			accessibilityProvider: {
+				getAriaLabel: (element) => element.text,
+				getWidgetAriaLabel: () => nls.localize('selectBox', "Select Box"),
+				getRole: () => 'option',
+				getWidgetRole: () => 'listbox'
 			}
 		});
 
 		this.values = this._options.values;
-
-		this._controller.onSelectionChange(e => {
-			this.value = e.value;
-			this._onValueChange.fire(e.value);
-			this._input.focus();
+		this._register(this._selectList.onDidBlur(() => {
 			this._hideList();
-		});
+		}));
 
-		this._controller.onDropdownEscape(() => {
-			this._hideList();
-			// have to put this in the setTimeout to make sure the focus can be set properly when the context menu is opened by pressing the DownArrow key
-			setTimeout(() => {
-				this._input.focus();
-			}, 0);
-		});
+		this._register(this._selectList.onKeyDown((e) => {
+			const event = new StandardKeyboardEvent(e);
+			let handled: boolean = false;
+			switch (event.keyCode) {
+				case KeyCode.Escape:
+					this._hideList();
+					setTimeout(() => {
+						this._input.focus();
+					}, 0);
+					handled = true;
+					break;
+				case KeyCode.Enter:
+				case KeyCode.Space:
+					const focusedElements = this._selectList.getFocusedElements();
+					if (focusedElements.length !== 0) {
+						this._updateSelection(focusedElements[0].text);
+						handled = true;
+					}
+					break;
+				default:
+					return;
+			}
+			if (handled) {
+				e.preventDefault();
+				e.stopPropagation();
+			}
+		}));
+		this._register(this._selectList.onMouseClick((e) => {
+			if (e.element) {
```

Comment:
Is this also closing the list too?

Reply:
yes, after you select an element, the list should be closed.